cli/azd/cmd/env.go

```diff
@@ -244,29 +240,34 @@ func envRefreshCmd(rootOptions *commands.GlobalCommandOptions) *cobra.Command {
 			return fmt.Errorf("loading environment: %w", err)
 		}
 
-		template, err := bicep.Compile(ctx, bicepCli, filepath.Join(azdCtx.InfrastructureDirectory(), "main.bicep"))
+		prj, err := project.LoadProjectConfig(azdCtx.ProjectPath(), &environment.Environment{})
 		if err != nil {
-			return err
+			return fmt.Errorf("loading project: %w", err)
 		}
 
-		res, err := azCli.GetSubscriptionDeployment(ctx, env.GetSubscriptionId(), env.GetEnvName())
-		if errors.Is(err, azcli.ErrDeploymentNotFound) {
-			return fmt.Errorf("no deployment for environment '%s' found. Have you run `infra create`?", rootOptions.EnvironmentName)
-		} else if err != nil {
-			return fmt.Errorf("fetching latest deployment: %w", err)
-		}
+		formatter := output.GetFormatter(ctx)
+		writer := output.GetWriter(ctx)
 
-		template.CanonicalizeDeploymentOutputs(&res.Properties.Outputs)
-		if err = saveEnvironmentValues(res, env); err != nil {
-			return err
+		infraManager, err := provisioning.NewManager(ctx, env, prj.Path, prj.Infra, !rootOptions.NoPrompt)
+		if err != nil {
+			return fmt.Errorf("creating provisioning manager: %w", err)
 		}
 
-		formatter, err := output.GetFormatter(cmd)
+		scope := provisioning.NewSubscriptionScope(ctx, env.GetLocation(), env.GetSubscriptionId(), env.GetEnvName())
+
+		getDeploymentResult, err := infraManager.GetDeployment(ctx, scope)
 		if err != nil {
+			return fmt.Errorf("getting deployment: %w", err)
+		}
+
+		if err := provisioning.UpdateEnvironment(&env, &getDeploymentResult.Deployment.Outputs); err != nil {
 			return err
 		}
+
+		console.Message(ctx, "Azure env refresh completed")
```

Comment:
```suggestion
		console.Message(ctx, "Environments setting refresh completed")
```

If we go by the verbiage used in the command:
"Refresh environment settings by using information from a previous infrastructure provision.",